planner/cascades/optimize.go

```diff
@@ -105,26 +114,29 @@ func findMoreEquiv(g *memo.Group, elem *list.Element) (eraseCur bool, err error)
 		// the transformation rule to enumerate all the possible expressions.
 		iter := memo.NewExprIterFromGroupElem(elem, pattern)
 		for ; iter != nil && iter.Matched(); iter.Next() {
+			// The pattern only identifies the operator type, some transformation rules also need
+			// detailed information for certain plan operators to decide whether it is applicable.
 			if !rule.Match(iter) {
 				continue
 			}
 
-			newExpr, erase, err := rule.OnTransform(iter)
+			newExprs, erase, _, err := rule.OnTransform(iter)
```

Comment:
We will use it when adding rules which do works like https://github.com/pingcap/tidb/blob/1805d7aa01bbe75c7a50602a322fd42b41bd8e71/planner/core/stats.go#L133. Hmm, I think `TODO` is unnecessary since we have added comments in `OnTransform` method of `Transformation`?